Classes/Service/SiteService.php

```diff
@@ -61,10 +62,40 @@ public function getFrontendUrl(string $url, int $pageUid): string
             if ($frontendBaseUrl !== '') {
                 $frontendBase = GeneralUtility::makeInstance(Uri::class, $this->sanitizeBaseUrl($frontendBaseUrl));
                 $frontBase = $frontendBase->getHost();
+                $frontPort = $frontendBase->getPort();
 
                 if (is_int(strpos($url, $base))) {
                     $url = str_replace($base, $frontBase, $url);
                 }
+
+                if ($port !== $frontPort) {
```

Comment:
Hi @kitzberger, 
thank you for contribution!

I tested your solution, it is working fine, but I think, we could simplify it a little bit like this:
replace lines with 71-98
```
if ($port === $frontPort) {
                    return $url;
                }

                $url = str_replace(
                    $frontBase . ($port ? ':'. $port : ''),
                    $frontBase . ($frontPort ? ':'. $frontPort : ''),
                    $url
                );
```
What I tested it working it all your cases. Could you confirm & replace it if possible, so we can merge your PR?